src/commands/periscope/helpers/periscopehelper.ts

```diff
@@ -109,51 +115,67 @@ export async function getStorageInfo(
     }
 }
 
+async function extractContainerName(clusterKubeConfig: string): Promise<string | undefined> {
+        const kubectl = await k8s.extension.kubectl.v1;
+
+        if (!kubectl.available) {
+            vscode.window.showWarningMessage(`Kubectl is unavailable.`);
+            return;
+        }
+
+        // Run cluster-info to get DNS Core hostname.
+        const runCommandResult = await tmpfile.withOptionalTempFile<k8s.KubectlV1.ShellResult | undefined>(
+            clusterKubeConfig, "YAML",
+            (f) => kubectl.api.invokeCommand(`cluster-info --kubeconfig="${f}"`));
+
+        // Get DNS Core hostname which Periscope use it as name of the container.
+        const matches = runCommandResult?.stdout.match(/(https?:\/\/[^\s]+)/g);
+        let containerName: string;
+        if (matches![0].indexOf('://') !== -1) {
```

Comment:
Again, how are you validating this?  You specifically allow `matches` to be undefined through null propagation, but then skip that possibility here.

I would suggest that if `runCommandResult` is undefined, you should detect that before this stanza, and _return a meaningful error_ (or return undefined if there is no meaningful error).  Proceed with line 132 _only after_ you have validated `runCommandResult` to be a ShellResult (and, I would suggest, checking that it succeeded via the exitCode).